python-sdk/src/astro/__init__.py

```diff
@@ -32,10 +31,3 @@ def get_provider_info() -> dict:
         "hook-class-names": [],
         "extra-links": [],
     }
-
-
-if not conf.getboolean(section="core", key="enable_xcom_pickling"):
-    raise OSError(
```

Comment:
@kaxil, what if we didn't raise an `OSError` but had a warning alerting users that some features may not work if `enable_xcom_pickling` is False? (e.g. `aql.dataframe`)? It can be misleading to not have any warning, and several features do not work.

I understand the motivation in allowing users to install and use the parts worth it without pickling - but, still, it's better to have a warning (at the very list on the tasks/decorators which will not work, such as `aql.dataframe`).

Reply:
`aql.dataframe` has a check before pushing to XCom though:

https://github.com/astronomer/astro-sdk/blob/b9c974ed105a4b0a95c09a17a055410ee0ad29c5/python-sdk/src/astro/sql/operators/dataframe.py#L166-L171

Reply:
where would you like me to add the warning